### PR TITLE
Get rid of input_driver_grab_mouse/input_driver_ungrab_mouse

### PR DESCRIPTION
**Pull request by twinaphex**
Fairly straightforward PR. Gets rid of these two getter functions.

### CODE REVIEW

File: retroarch.c [hunk 1]
```diff
@@ -8382,15 +8387,30 @@ bool command_event(enum event_command cmd, void *data)
             {
                video_driver_hide_mouse();
                if (!settings->bools.video_windowed_fullscreen)
-                  if (input_driver_grab_mouse())
+               {
+                  if ((input_st->current_driver &&
```
Comment (jdgleaver) [inline retroarch.c:8391]: Unnecessary double brackets

File: retroarch.c [hunk 2]
```diff
@@ -8561,13 +8581,23 @@ bool command_event(enum event_command cmd, void *data)
 
             if (grab_mouse_state)
             {
-               if ((ret = input_driver_grab_mouse()))
+               if ((ret = input_st->current_driver &&
+input_st->current_driver->grab_mouse))
```
Comment (jdgleaver) [inline retroarch.c:8585]: Wrong indentation
Unnecessary double brackets

File: retroarch.c [hunk 3]
```diff
@@ -8561,13 +8581,23 @@ bool command_event(enum event_command cmd, void *data)
...
+               {
+                  input_st->current_driver->grab_mouse(
+                        input_st->current_data, true);
                   input_st->grab_mouse_state = true;
+               }
             }
             else
             {
-               if ((ret = input_driver_ungrab_mouse()))
+               if ((ret = input_st && input_st->current_driver &&
```
Comment (jdgleaver) [inline retroarch.c:8594]: This is broken. `ret` was previously being used to check whether `input_driver_grab_mouse()`. What you have copy/pasted here is fundamentally different (and will not work)

File: retroarch.c [hunk 4]
```diff
@@ -8561,13 +8581,23 @@ bool command_event(enum event_command cmd, void *data)
...
+                     input_st->current_driver->grab_mouse))
```
Comment (jdgleaver) [inline retroarch.c:8595]: Unnecessary double brackets

File: retroarch.c [hunk 5]
```diff
@@ -8660,16 +8690,26 @@ bool command_event(enum event_command cmd, void *data)
 
                if (input_st->game_focus_state.enabled)
                {
-                  if (input_driver_grab_mouse())
+                  if ((input_st->current_driver &&
```
Comment (jdgleaver) [inline retroarch.c:8693]: Unnecessary double brackets

File: retroarch.c [hunk 6]
```diff
@@ -18410,15 +18450,27 @@ static bool video_driver_init_internal(
    if (input_st->grab_mouse_state)
    {
       video_driver_hide_mouse();
-      if (input_driver_grab_mouse())
+      if ((input_st->current_driver &&
```
Comment (jdgleaver) [inline retroarch.c:18453]: Unnecessary double brackets

File: retroarch.c [hunk 7]
```diff
@@ -18410,15 +18450,27 @@ static bool video_driver_init_internal(
...
          input_st->grab_mouse_state = true;
+      }
    }
    else if (video.fullscreen)
    {
       video_driver_hide_mouse();
       if (!settings->bools.video_windowed_fullscreen)
-         if (input_driver_grab_mouse())
+      {
+         if ((input_st->current_driver &&
```
Comment (jdgleaver) [inline retroarch.c:18466]: Unnecessary double brackets

**Comment by twinaphex** [comment]
OK, thanks for the feedback. I'll nuke this PR and start over from scratch.

**Pull request closed without merging**